src/components/FormModal/FormModal.js

```diff
@@ -92,27 +98,100 @@ export default function FormModal({ onCancelAdding, id, name }) {
   }
 
   function handleMorningClicked(name) {
-    const newMorningChecked = weekRoutine.days.map((day) => {
-      if (day.name === name) {
-        day.morning = !day.morning;
-        return day;
-      } else {
-        return day;
-      }
+    const i = getIndexForWeekday(name);
+    const checkedMornings = routineData.filter(
+      (product) =>
+        product.days[i].name === name && product.days[i].morning === true
+    );
+
+    const routineIDs = checkedMornings.map((product) => product.id);
+    const intersection = routineIDs.filter((id) => conflicts.includes(id));
+
+    const conflictName = intersection.map((id) => {
+      const conflictProduct = getProductById(id, products);
+      return conflictProduct.name;
     });
-    setWeekRoutine({ id: id, days: newMorningChecked, date: weekRoutine.date });
+
+    if (intersection.length > 0) {
+      alert(
+        `You are already using ${conflictName} on ${name} morning. These two products have conflicting ingredients`
+      );
+    } else {
+      const newMorningChecked = weekRoutine.days.map((day) => {
+        if (day.name === name) {
+          day.morning = !day.morning;
+          return day;
+        } else {
+          return day;
+        }
+      });
+      setWeekRoutine({
+        id: id,
+        days: newMorningChecked,
+        date: weekRoutine.date,
+      });
+    }
   }
 
   function handleEveningClicked(name) {
-    const newEveningChecked = weekRoutine.days.map((day) => {
-      if (day.name === name) {
-        day.evening = !day.evening;
-        return day;
-      } else {
-        return day;
-      }
+    const i = getIndexForWeekday(name);
+    const checkedEvenings = routineData.filter(
+      (product) =>
+        product.days[i].name === name && product.days[i].evening === true
+    );
+
+    const routineIDs = checkedEvenings.map((product) => product.id);
+    const intersection = routineIDs.filter((id) => conflicts.includes(id));
+
+    const conflictName = intersection.map((id) => {
+      const conflictProduct = getProductById(id, products);
+      return conflictProduct.name;
+    });
+
+    if (intersection.length > 0) {
+      alert(
+        `You are already using ${conflictName} on ${name} evening. These two products have conflicting ingredients`
+      );
+    } else {
+      const newEveningChecked = weekRoutine.days.map((day) => {
+        if (day.name === name) {
+          day.evening = !day.evening;
+          return day;
+        } else {
+          return day;
+        }
+      });
+      setWeekRoutine({
+        id: id,
+        days: newEveningChecked,
+        date: weekRoutine.date,
+      });
+    }
+  }
```

Comment:
This seems to be code you had already written. Maybe you can create a util or service file which you can call to do this stuff, so you don't have to write the code everytime you need the logic.